enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,1050 @@
+  so a compromised component cannot break my cluster (too much).
+  * Still, CSI driver Pods *must* run on each worker node in the guest cluster as privileged, to be able to format and
+    mount volumes. A compromised CSI driver Pod can do anything to the application data - steal it, delete it, encrypt
+    it..., and it can fully compromise the node.
+    
+* As a Cluster Service Consumer, I want to spawn new OCP guest clusters on AWS, so the guest clusters can use AWS EBS
+  storage out of the box.
+  * Explicitly, the guest clusters can run in different AWS regions and zones than the HyperShift management cluster.
+    The management cluster does not need to run on AWS at all. In addition, one management cluster may run guest
+    clusters in different AWS regions or even in different clouds.
+
+* As a Cluster Service Consumer, I want my OCP guest clusters with no worker nodes, so I can 1) autoscale the cluster
+  from zero or 2) "hibernate" the cluster when it's idle and resume it later.
+  * Control-plane components should keep working without any issues.
+    * Cluster Instance Users should be e.g. able to delete PVCs and expect that the corresponding volume is deleted in AWS.
+    * Dynamic provisioning should work if forced by `volumeBindingMode: Immediate`. Users are then responsible
+      for creating nodes in the right availability zones so that they can use the provisioned volumes.
+      
+* Phase 2: As a Cluster Service Consumer, I want to get metrics of storage management cluster components, so I can
+  monitor them.
+  * This is Phase 2, as most useful storage metrics are already emitted by kube-controller-manager. While metrics of the
+    CSI driver, its sidecars, snapshot-controller and their operators are interesting, they're not crucial.
+
+* Phase 2: As a Cluster Service Consumer, I want all management cluster components that talk to a guest cluster to have
+  proper RBACs, so a compromised storage component can't break the guest cluster (too much).
+  * The initial implementation may give cluster-admin role in the guest cluster to storage components in the management
+    cluster - a compromised component in the management cluster could get full access the guest cluster, but not to the
+    management cluster.
+  * Even with the minimal RBACs, CSI driver components will be able to delete any PV and any volume in the cloud,
+    messing up the guest cluster pretty badly.
+
+* As a Cluster Service Provider, I want the control plane for guest clusters to be as small as possible, so I can run
+  many guest clusters in my management clusters.
+  * Especially AWS EBS CSI driver control-plane Deployment is taking ~300 MiB of memory, because it runs too many
+    sidecars + RBAC proxies.
+
+* As a Cluster Service Provider, I want guest clusters without any AWS credentials, so Cluster Instance Admins can't
+  mis-use the credentials to either run something out of the guest cluster or break the cloud objects.
```

Comment:
@derekwaynecarr @csrwng I just added this story because I think we talked about it, but [HyperShift goals and design invariants](https://hypershift-docs.netlify.app/reference/goals-and-design-invariants/) does not mention anything about cloud credential. Is it a real requirement then?

Reply:
It is a real requirement and we should at it to the doc that you linked to.

Reply:
the credentials to access cloud objects are sealed on the mgmt cluster (not visible in the guest).  trust segmentation is real requirement.

Reply:
Ok, I'll leave the story there.